drivers/hmis/app/models/hmis/ce/match/candidate_pool.rb

```diff
@@ -41,13 +41,21 @@ class CandidatePool < GrdaWarehouseBase
     # foreign key constraint violations
     before_destroy :nullify_deleted_associations
 
-    # pools for active opportunities
     scope :active, -> {
-      active_ids = ::Hmis::Ce::Opportunity.active.pluck(:candidate_pool_id).compact.uniq
-      where(id: active_ids)
+      # Pool is active if there are any active Opportunities that reference it
+      active_ids_for_opportunities = ::Hmis::Ce::Opportunity.active.pluck(:candidate_pool_id).compact.uniq
+      # Pool is active if there are any UnitGroups that reference it
+      active_ids_for_unit_groups = Hmis::UnitGroup.with_ce_waitlists_enabled.pluck(:candidate_pool_id).compact.uniq
+
+      where(id: active_ids_for_opportunities + active_ids_for_unit_groups)
```

Comment:
small optimization: could use distinct on the queries and unique on the id set 
```suggestion
      active_ids_for_opportunities = ::Hmis::Ce::Opportunity.active.distinct.pluck(:candidate_pool_id).compact
      # Pool is active if there are any UnitGroups that reference it
      active_ids_for_unit_groups = Hmis::UnitGroup.with_ce_waitlists_enabled.distinct.pluck(:candidate_pool_id).compact
      all_active_ids = active_ids_for_opportunities + active_ids_for_unit_groups).sort.uniq
      where(id: all_active_ids)
```